ccmlib/node.py

```diff
@@ -530,9 +530,19 @@ def watch_log_for_alive(self, nodes, from_mark=None, timeout=120, filename='syst
         Watch the log of this node until it detects that the provided other
         nodes are marked UP. This method works similarly to watch_log_for_death.
         """
-        tofind = nodes if isinstance(nodes, list) else [nodes]
-        tofind = [f"({node.address()}|{node.hostid()}).* now UP" for node in tofind]
-        self.watch_log_for(tofind, from_mark=from_mark, timeout=timeout, filename=filename)
+        patterns = []
+        for node in nodes if isinstance(nodes, list) else [nodes]:
+            host_id = None
+            try:
+                host_id = node.hostid()
+            except Exception:
+                pass
+            if not host_id:
+                # If failed to get host_id, replate it with a pattern,
+                # Address is enough for node identification
```

Comment:
What does it mean? There is no longer address reported in the "now UP" message so as far as I see this will simply match UP message for any node.

Reply:
Since when ? I see the following on 2025.1:
```
INFO  2025-04-24 01:02:07,908 [shard 0: gms] gossip - InetAddress 62671905-d3b4-4189-bedf-cd2f02926bea/127.0.0.2 is now UP, status = UNKNOWN
```

I have modified the code to look for ip-address and node-id when node-id is available, otherwise only ip-address.

Reply:
Ah, it was dropped for DOWN message, but not for UP. Strange.